application/asset/js/site-item-sets.js

```diff
@@ -21,46 +28,51 @@ var appendItemSet = function(id, title, email) {
     $('#site-item-set-rows').append(row);
 }
 
+var updateItemSetCount = function(itemSetId) {
+    var itemSet = $('[data-item-set-id="' + itemSetId + '"]');
+    var itemSetParent = itemSet.parents('.selector-parent');
+    var childCount = itemSetParent.find('.selector-child-count').first();
+    var newTotalCount = parseInt(totalCount.text()) - 1;
+    var newChildCount = parseInt(childCount.text()) - 1;
+    totalCount.text(newTotalCount);
+    childCount.text(newChildCount);
+}
+
 if (itemSetsData.length) {
     $.each(itemSetsData, function() {
         appendItemSet(this.id, this.title, this.email);
+        $('[data-item-set-id="' + this.id + '"]').hide();
+        updateItemSetCount(this.id);
     });
-    itemSets.show();
-} else {
-    noItemSets.show();
+    $('#item-sets-section').addClass('has-item-sets');
 }
 
-$('#item-set-selector .selector-child').on('click', function() {
+$('#item-set-selector .selector-child').on('click', function(e) {
+    e.stopPropagation();
     var itemSet = $(this);
+    var itemSetParent = itemSet.parents('.selector-parent');
+    itemSetParent.unbind('click');
     appendItemSet(
         itemSet.data('itemSetId'),
         itemSet.data('childSearch'),
         itemSet.data('ownerEmail')
     );
-    itemSets.show();
-    noItemSets.hide()
+    itemSet.hide();
+    updateItemSetCount(itemSet.data('itemSetId'));
+    itemSetParent.bind('click', parentToggle);
+    Omeka.scrollTo($('.site-item-set-row:last-child'));
 });
 
-itemSets.on('click', '.o-icon-delete', function(e) {
+itemSets.on('click', '.o-icon-delete, .o-icon-undo', function(e) {
     e.preventDefault();
-    var removeLink = $(this);
-    var undoLink = removeLink.siblings('.o-icon-undo');
-    var row = removeLink.closest('.site-item-set-row');
-    row.find('input').prop('disabled', true);
-    row.toggleClass('delete');
-    removeLink.hide();
-    undoLink.show();
-});
-
-itemSets.on('click', '.o-icon-undo', function(e) {
-    e.preventDefault();
-    var undoLink = $(this);
-    var removeLink = undoLink.siblings('.o-icon-delete');
-    var row = undoLink.closest('.site-item-set-row');
-    row.find('input').prop('disabled', false);
+    var row = $(this).closest('.site-item-set-row');
+    var hiddenInput = row.find('input');
+    if (hiddenInput.prop('disabled') !== true) {
```

Comment:
This whole if/else can just be `hiddenInput.prop('disabled', !hiddenInput.prop('disabled'))`